### DISCUSSION TITLE
SMILES <-> Molfiles?

### DISCUSSION THREAD

**Discussion by phytophreak** [rdkit/rdkit#6813] [Q&A]
Hi,

I am currently trying to merge some external compounds with the compounds present in the ChEMBL database. My approach is to use the InChIKey as unique identifier. For the ChEMBL compounds, this information is already available. But for new compounds it first needs to generated from some “initial” format, which is usually SMILES. I have understood that the “source of truth” in the ChEMBL database are the Molfiles and SMILES are just “secondary”. But the problem is that that for many other datasets I initially only have SMILES as representation. And Molfiles first need to be generated from those SMILES. 

To narrow the problem down a bit more: 
Depending on whether you start with Molfiles or SMILES, the resulting InChIKeys might be different. One example from the ChEMBL database itself:

The SMILES string and Molblock were taken from this compound: https://www.ebi.ac.uk/chembl/compound_report_card/CHEMBL2391680/
`Chem.MolToInchiKey(Chem.MolFromSmiles(…)) = 'RNSAWGNXDCNAAY-UHFFFAOYSA-N'`
`Chem.MolToInchiKey(Chem.MolFromMolBlock(…)) = 'RNSAWGNXDCNAAY-DXDUWVORSA-N'` (this is equal to the one in the ChEMBL database)

We can see that the InChIKeys are different. 

My first question: Is there even a way to arrive at the same InChIKey if we start from SMILES as input? So basically can we go back and forth between SMILES and Molfiles?
And the second question: If this is not possible, what is the best way to deal with new data where we initially don’t have a Molfile? 

I used RDKit version 2023.3.3 (but tried also a few other versions).
A question related to this problem was already asked on GitHub in the `chembl_structure_pipeline` repo: https://github.com/chembl/ChEMBL_Structure_Pipeline/issues/53

Thank you for your help!

Kind regards,
Bernhard

**Comment by schatzsc** [comment]
The InChIKey is NOT a "unique identifier" - only the InChI string strictly fulfills the condition "Only one identifier for each unique structure and no identifier used more than one time"

In some rare cases, two compounds with different InChI identifier strings can still get assigned identical InChIKeys as a result of information lost in the course of the "hashing" processing on the InChI string:

https://www.inchi-trust.org/
